services/app-api/src/resolvers/unlockHealthPlanPackage.ts

```diff
@@ -129,16 +130,30 @@ export function unlockHealthPlanPackageResolver(
             throw new Error(errMessage)
         }
 
-        const programs = store.findPrograms(draft.stateCode, draft.programIDs)
-        if (!programs || programs.length !== draft.programIDs.length) {
-            const errMessage = `Can't find programs ${draft.programIDs} from state ${draft.stateCode}, ${draft.id}`
+        const isContractAndRate = draft.submissionType === 'CONTRACT_AND_RATES'
+
+        const combinedProgramIDs = [...draft.programIDs]
+        if (isContractAndRate && draft.rateProgramIDs) {
```

Comment:
Could we just adjust our store function tho return ALL possible programs for the state if no program ids are passed in?  I think that might cut down on complexity here in the resolver, we wouldn't need to generate that specific program ids list here and in each resolver that needs it. 

Instead,  we could just do that filtering and error handling logic in one place- in  the  shared`generateRateName` utility. Just pass in all available programs for the state each time.

Reply:
We are also checking to make sure the length of the `programs` is the same as the `combinedPrograms`. I believe this was done to make sure we could find all the program names and throw and error if not.

```
    if (!programs || programs.length !== combinedProgramIDs.length) {
        const errMessage = `Can't find programs ${combinedProgramIDs} from state ${draft.stateCode}, ${draft.id}`
        logError('unlockHealthPlanPackage', errMessage)
        setErrorAttributesOnActiveSpan(errMessage, span)
        throw new Error(errMessage)
    }
```

Reply:
I think we could make a return all programs, but it would require more refactoring than expected. 

We would still need to check to make sure if programs from the `store` and programs from the package matched in length, otherwise we would get `Unknown Program` when generating names. 

We are also throwing this error before updating the db, moving this would still update the db, but not send emails on errors.

```
        if (!programs || programs.length !== combinedProgramIDs.length) {
            const errMessage = `Can't find programs ${combinedProgramIDs} from state ${draft.stateCode}, ${draft.id}`
            logError('unlockHealthPlanPackage', errMessage)
            setErrorAttributesOnActiveSpan(errMessage, span)
            throw new Error(errMessage)
        }
```

```
if (!programs || programs.length !== combinedProgramIDs.length) {
```

It doesn't seem explicit that this checks to make sure we have matching programs, but `findPrograms()` filters out state programs from our `statePrograms.json` that are not included in programs from the submission. If the length does not match, this means a program there was an issue finding a specific program from either the submission or `statePrograms.json` which would cause errors generating names.